packages/beeq/src/components/slider/bq-slider.tsx

```diff
@@ -203,6 +224,28 @@ export class BqSlider {
     return (value / totalRange) * 100;
   };
 
+  // The below calculations are used to "guess" where the thumb is located. Since we're using the native range control
+  // under the hood, we don't have access to the thumb's true coordinates. These measurements can be a pixel or two
+  // off depending on the size of the control, thumb, and tooltip dimensions.
+
+  private calculateThumbPosition = (): { leftThumbPosition: number; rightThumbPosition?: number } => {
+    if (!this.progressElem || !this.inputDivElement) return;
+
+    // Get the width of the input container and the size of the slider thumb
+    const inputWidth = this.inputDivElement.getBoundingClientRect().width;
+    const thumbSize = parseInt(getComputedStyle(this.progressElem).getPropertyValue('--bq-slider--thumb-size'), 10);
+    const { min, max } = this;
+    const totalInputWidth = inputWidth - thumbSize;
+
+    // Define and calculate the thumb positions
+    const calculatePosition = (val: number): number => ((val - min) / (max - min)) * totalInputWidth + thumbSize / 2;
+
+    const leftThumbPosition = calculatePosition(Number(this.minValue));
+    const rightThumbPosition = this.isRangeType ? calculatePosition(Number(this.maxValue)) : undefined;
+
+    return { leftThumbPosition, rightThumbPosition };
+  };
```

Comment:
Perhaps we can move this below the `updateProgressTrack()` method. Also I will recommend to split a bit the logic here, something like the following could work maybe:

```ts
// The below calculations are used to "guess" where the thumb is located. Since we're using the native range control
// under the hood, we don't have access to the thumb's true coordinates. These measurements can be a pixel or two
// off depending on the size of the control, thumb, and tooltip dimensions.

private calculateThumbPosition = (value: number): number => {
  if (!this.trackElem) return;

  // Get the width of the track area and the size of the slider thumb
  const inputWidth = this.trackElem.getBoundingClientRect().width;
  const thumbSize = parseInt(getComputedStyle(this.el).getPropertyValue('--bq-slider--thumb-size'), 10);
  const totalInputWidth = inputWidth - thumbSize;

  return ((value - this.min) / (this.max - this.min)) * totalInputWidth + thumbSize / 2;
};

private thumbPosition = (): { leftThumbPosition: number; rightThumbPosition?: number } => {
  const leftThumbPosition = this.calculateThumbPosition(this.minValue);
  const rightThumbPosition = this.isRangeType ? this.calculateThumbPosition(this.maxValue) : undefined;

  return { leftThumbPosition, rightThumbPosition };
};
```

Notice how we can use `this.el` to get the thumb size CSS property value, at the end, they are all declared at the `host` level.
On the same topic, `this.minValue` and `this.maxValue` have `number` type in the state declaration, no need to coerce them again 😉 .